salt/state.py

```diff
@@ -2305,7 +2305,10 @@ def __gen_opts(self, opts):
                 'state_auto_order',
                 opts['state_auto_order'])
             opts['file_roots'] = mopts['file_roots']
-            opts['env_order'] = mopts['env_order']
+            opts['top_file_merging_strategy'] = mopts.get('top_file_merging_strategy',
```

Comment:
@cachedout, is this the correct way to prefer master opts, with a fallback to minion opts, for example to handle the masterless case?